packages/typegpu/src/data/wgslTypes.ts

```diff
@@ -330,11 +330,47 @@ interface Swizzle4<T2, T3, T4> extends Swizzle3<T2, T3, T4> {
   readonly wwww: T4;
 }
 
+interface Tuple2 {
```

Comment:
these interface names can be confusing. maybe something like `WithAsTupleX`? `TuplableX`?